### PR TITLE
[FLINK-8475][config][docs] Integrate RM options

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR integrates the RM `ConfigOptions` into the configuration docs generator.

## Brief change log

* Add missing descriptions to config options (derived from existing description/javadocs)
* integrate RM configuration table into `config.md`

### CODE REVIEW

File: docs/_includes/generated/resource_manager_configuration.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            <td>1</td>
+            <td></td>
+        </tr>
+        <tr>
+            <td><h5>resourcemanager.job.timeout</h5></td>
+            <td>"5 minutes"</td>
+            <td>Timeout for jobs which don't have a job manager as leader assigned.</td>
+        </tr>
+        <tr>
+            <td><h5>resourcemanager.rpc.port</h5></td>
```
Comment (aljoscha) [inline docs/_includes/generated/resource_manager_configuration.html]: Why does this not have the description from the original doc?

Comment (zentol) [reply docs/_includes/generated/resource_manager_configuration.html]: Must've missed it by accident, added it back.

**Comment by aljoscha** [comment]
Good to go 👌

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**